dual_net.py

```diff
@@ -134,6 +139,45 @@ def should_log(i):
                     logger.add_summary(weight_summaries, global_step)
             self.save_weights()
 
+    def validate(self, tf_records, batch_size=128, init_from=None, num_steps=1000):
+        """Compute only the cost components for a set of tf_records, ideally a
+        holdout set, and report them to an 'eval' subdirectory of the logs.
+        """
+        cost_tensor_names = ['policy_cost', 'value_cost', 'l2_cost',
+                             'combined_cost']
+        if self.logdir is None:
+            print("Error, trainer not initialized with a logdir.")
+            return
+
+        logdir = self.logdir + '-eval'
+
+        with self.sess.graph.as_default():
+            input_tensors = preprocessing.get_input_tensors(
+                batch_size, tf_records, shuffle_buffer_size=10000, filter_amount=0.5)
```

Comment:
What is this filter amount constant?

Reply:
I'm also afraid of this filter constant. What if our eval results are different for differing values of this constant?

Reply:
i'm not sure why they would be different: we aren't going to overfit to them ;P  Our holdout data is already small -- 5% of all games.  Taking 5% of 5% might be very few positions, and running for 1k steps will iterate repeatedly if they are exhausted, [as we call read_tf_records with num_repeats=None everywhere](https://github.com/tensorflow/minigo/blob/master/preprocessing.py#L137), which would lead to *noisier* eval results if we *don't* step up this filter amount.

Josh: i'm not sure if you want a comment on it here -- it's commented in preprocessing.py.